syntax = "proto3";

import "google/protobuf/timestamp.proto";

message SensorMessage {
    oneof message_type {
        BearingMessage bearing = 1;
        PositionMessage position = 2;
        PoseMessage pose = 3;
    }
}

message BearingMessage {
    double bearing = 1;
    google.protobuf.Timestamp timestamp = 2;
}

message PositionMessage {
    double x = 1;
    double y = 2;
    google.protobuf.Timestamp timestamp = 3;
}

message PoseMessage {
    double x = 1;
    double y = 2;
    double heading = 3;
    google.protobuf.Timestamp timestamp = 4;
}
